src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -80,6 +81,16 @@ private void OnStatusChanged(bool started)
             return await connection.TryInvokeAsync(invocation, cancellationToken).ConfigureAwait(false);
         }
 
+        public async IAsyncEnumerable<TResult> TryInvokeStreamAsync<TService, TResult>(
+            Func<TService, CancellationToken, IAsyncEnumerable<TResult>> invocation,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+            where TService : class
+        {
+            using var connection = CreateConnection<TService>(callbackTarget: null);
+            await foreach (var result in connection.TryInvokeStreamAsync(invocation, cancellationToken).WithCancellation(cancellationToken))
+                yield return result;
```

Comment:
@AArnott This was the part i was originally scared above.  but it turned out to be trivial.  i had thought i was going to return the ienumerable and then dispose the connection.  But like with any enumerable, i'm streaming the items, then cleaning up after that.